src/python/pants/engine/legacy/structs.py

```diff
@@ -359,8 +359,10 @@ def filespecs(self) -> wrapped_globs.Filespec:
   def to_path_globs(self, relpath: str, conjunction: GlobExpansionConjunction) -> PathGlobs:
     """Return a PathGlobs representing the included and excluded Files for these patterns."""
     return PathGlobs(
-      include=tuple(os.path.join(relpath, glob) for glob in self._parsed_include),
-      exclude=tuple(os.path.join(relpath, exclude) for exclude in self._parsed_exclude),
+      globs=(
+        *(os.path.join(relpath, glob) for glob in self._parsed_include),
+        *(f"!{os.path.join(relpath, glob)}" for glob in self._parsed_exclude)
+      ),
```

Comment:
https://github.com/pantsbuild/pants/pull/9025 will follow this up to add explicit support for the `sources` field.

Reply:
Forgive me for being incredibly lazy, but what explicit support for the `sources` field are you referring to in this case? It's not immediately obvious from checking this file in #9025.